### PR TITLE
Add snapshot user registry

### PR DESCRIPTION
**Pull request by yuetloo**
This user registry will add users who can be verified by the proof of token ownership at a snapshot block.

### CODE REVIEW

File: contracts/contracts/userRegistry/SnapshotUserRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    minBalance = newMinBalance;
+
+    emit MinBalanceChanged(minBalance);
+  }
+
+  /**
+   * @dev Reject user
+    * @param _user user account address
+   */
+  function rejectUser(address _user) external onlyOwner {
```
Comment (auryn-macmillan) [inline contracts/contracts/userRegistry/SnapshotUserRegistry.sol]: What are the purposes of `rejectUser` and `resetUser`?
How do we envision them being used in practice, @yuetloo?

File: contracts/tests/userRegistrySnapshot.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        accountProofRlpBytes
+      )
+      await tx.wait()
+    } catch (err) {
+      console.log('error setting storage hash', err)
+      throw err
+    }
+  })
+
+  describe('Add user', function () {
```
Comment (auryn-macmillan) [inline contracts/tests/userRegistrySnapshot.ts:91]: Would be good to add tests for users that currently meet the requirements, but did not at the snapshot block, and for users who do not currently meet the requirements, but did at the snapshot block.

File: contracts/contracts/userRegistry/SnapshotUserRegistry.sol (hunk 1 above)
Comment (yuetloo) [reply contracts/contracts/userRegistry/SnapshotUserRegistry.sol]: I was thinking to let the coordinator reject sybil accounts. But, you're right that in practice, it's too late as once the user is added, they would've contributed and in the round.

**Comment by yuetloo** [comment]
Closing this PR as changes have been merged to https://github.com/clrfund/monorepo/pull/711

**Pull request closed without merging**